default/scripting/ship_parts/targeting.macros

```diff
@@ -22,11 +22,12 @@ COMBAT_TARGETS_NOT_DESTROYED_SHIP
 '''
 
 COMBAT_TARGETS_PLANET_WITH_DEFENSE
-'''         And [
+'''        And [
             Planet
             Or [
-               Not Shield high = 0
-               Not Defense high = 0
+                Not Shield high = 0
+                Not Defense high = 0
+                Not Construction high = 0
```

Comment:
Does construction somehow affect it?

Reply:
This way it mimics the way it is done in the backend. So the answer must be yes.

Reply:
Yes in the past it was too messy to combine regeneration for defense and shield for triggering combat with the usual balance and intended feature (zero defense and zero shields should be possible on a colony). So we reused the unused construction for triggering combat by ensuring minimum construction one (on colonies).